Makefile

```diff
@@ -67,6 +67,11 @@ test:
 		pytest --cov=$(PY_MODULE) test/ $(T) $(TEST_ARGS) && \
 		python -m coverage report -m $(COV_ARGS)
 
+	# Only generate an XML coverage report if explicitly requested.
+	[ -n "$(COV_REPORT)" ] && \
+		. ./env/bin/activate && \
+		python -m coverage xml || exit 0
```

Comment:
shell smell here: the `exit 0` is used to handle the "no coverage requested" here but it also hides potential coverage failures

I realize the reason is that Makefile sucks for shell scripts. Maybe fixing this is not worth the sea of "; /" that you likely  need to add to handle this in a real if-clause...

Reply:
This actually won't mask the coverage failure, since the previous line in the target does that 🙂 -- this last line is solely for the XML generation. But yeah, I agree the `|| exit 0` pattern is ugly; I'll refactor.